Improve blocks style on classic theme - editor side

gigitux opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

This PR improves how some blocks style on classic themes.

###  Ensure `Star` Font Loads in the Editor for Classic Theme

The `Star` font wasn’t loading in the editor when using the classic theme because the relevant CSS stylesheet wasn’t enqueued.

This PR introduces `woocommerce-classictheme-editor-fonts.css` and ensures it is enqueued on the editor side.

To guarantee the stylesheet loads within the iframe, it includes selectors like `.editor-styles-wrapper`, `.wp-block`, or `.wp-block-*`. More details can be found in [this comment from the Gutenberg codebase](https://github.com/WordPress/gutenberg/blob/7129676335fe56802978a9511d860865f7e65aec/packages/block-editor/src/components/iframe/get-compatibility-styles.js#L3-L10).

| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/5f529734-6595-4f60-86d0-c83a21b6485b) | ![image](https://github.com/user-attachments/assets/9449d032-b477-4b64-a053-fde5f81a4cde) | 


### Improve Product Filters layout

This PR enhances the Product Filters layout on classic themes by increasing CSS specificity:

https://github.com/woocommerce/woocommerce/pull/56677/files#diff-762bab41217351ef6ac94a5b5f240cd26ddb23253d9b6fe95f6046227d887a43R40-R41

On classic themes, WordPress enqueues [this CSS](https://github.com/joemcgill/wordpress-develop/blob/2c647d67081cf4246b2032a317e34ebfb83580c9/src/wp-admin/css/forms.css#L112-L133), which previously had higher specificity, affecting the filters' styling. This update ensures WooCommerce styles take precedence.  

| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/e24b3daf-31b4-4f72-b2bf-cc217f12a34a) | ![image](https://github.com/user-attachments/assets/570f6ea5-83ad-4fa7-8043-606a6063411a)| 




Closes #55842.





<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that you're using a classic theme like Storefront and you're on WordPress 6.8.
2. On the sidebar, Appearance -> Design -> Styles.
3. Compare all the blocks mentioned in the table in #55842 and ensure that they look good.


<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Improve blocks style on classic theme - editor side

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

gigitux committed:
Improve blocks style on classic theme - editor side

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks, woocommerce

gigitux commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/style.scss:
```diff
@@ -36,17 +37,21 @@
 	}
 }
 
+span.wc-block-product-filter-checkbox-list__input-wrapper
```
This is the only change that I did to this file

gigitux committed:
fix lint errors

gigitux committed:
Merge branch 'release/9.8' into 55842-wordpress-68-style-book-integration-with-classic-themes

roykho commented:
Hi @gigitux - sorry but how do you get to the style book if you're using Storefront?

gigitux commented:
> Hi @gigitux - sorry but how do you get to the style book if you're using Storefront?

Sorry, I forgot to mention that you need to use WordPress 6.8. I'm going to update the testing instructions.

roykho commented:
Am I correct that this PR does not cover the checkbox? Because currently it doesn't look right.
![CleanShot 2025-03-27 at 05 54 57](https://github.com/user-attachments/assets/0ff237cf-b173-44cd-8af1-7a247009b41c)

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into 55842-wordpress-68-style-book-integration-with-classic-themes

gigitux committed:
Merge branch '55842-wordpress-68-style-book-integration-with-classic-themes' of github.com:woocommerce/woocommerce into 55842-wordpress-68-style-book-integration-with-classic-themes

gigitux committed:
fix css

gigitux commented:
Sorry for the confusion! To fix the linter errors, I updated in the wrong way the CSS selector. I fixed it with c08c401952923dfcc8f2ae0fb8e0042d5a310c17.
It should be fine now!

roykho commented on plugins/woocommerce/includes/admin/class-wc-admin-assets.php:
```diff
@@ -62,6 +62,11 @@ public function admin_styles() {
 			wp_style_add_data( 'woocommerce_admin_privacy_styles', 'rtl', 'replace' );
 
 			if ( $screen && $screen->is_block_editor() ) {
+				if ( ! wc_current_theme_is_fse_theme() ) {
```
```suggestion
				if ( ! wp_is_block_theme() ) {
```

gigitux replied:
Updated with 105ebd5ec81150d58a3d862a23cba71f18761af1

roykho reviewed (approved):
Great work! Just left a small commit suggestion.

![CleanShot 2025-03-28 at 18 34 49](https://github.com/user-attachments/assets/d72f91e1-1280-405c-b9a4-33066ca68cf3)
![CleanShot 2025-03-28 at 18 35 00](https://github.com/user-attachments/assets/08727925-8f0e-407d-b4a3-32766c3d7c57)
![CleanShot 2025-03-28 at 18 35 16](https://github.com/user-attachments/assets/2c341394-2bd8-4e4f-8183-4c7eee181a62)
![CleanShot 2025-03-28 at 18 35 22](https://github.com/user-attachments/assets/60ad9793-f744-44c3-87b0-f437412bc879)
![CleanShot 2025-03-28 at 18 35 57](https://github.com/user-attachments/assets/fc6a6403-80fe-4834-9f48-11fcacd97de7)
![CleanShot 2025-03-28 at 18 36 17](https://github.com/user-attachments/assets/4e532e0c-4826-4eb1-8fc5-2444cf88f396)

gigitux committed:
update method

gigitux committed:
Merge branch 'trunk' into 55842-wordpress-68-style-book-integration-with-classic-themes

gigitux committed:
Merge branch 'trunk' into 55842-wordpress-68-style-book-integration-with-classic-themes

gigitux committed:
Merge branch 'trunk' into 55842-wordpress-68-style-book-integration-with-classic-themes

dinhtungdu committed:
Merge branch 'trunk' into 55842-wordpress-68-style-book-integration-with-classic-themes

dinhtungdu commented on plugins/woocommerce/client/blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/style.scss:
```diff
@@ -36,6 +36,7 @@
 	}
 }
 
+span.wc-block-product-filter-checkbox-list__input-wrapper
 :where(.wc-block-product-filter-checkbox-list__input) {
```
In this case, I think we can remove `:where` and use normal selector.
```suggestion
input.wc-block-product-filter-checkbox-list__input {
```

dinhtungdu reviewed (approved):
Thanks for working on this! I can confirm this PR fix block style for input and star for classic theme on WP 6.8. I left an optional comment on the CSS selector choice, but otherwise this is LGTM so I'm approving.

#### Screenshots

| Before | After |
|--------|--------|
| <img width="1381" alt="image" src="https://github.com/user-attachments/assets/178d55a4-25c8-4753-b802-4c05904247f7" /> | <img width="1381" alt="image" src="https://github.com/user-attachments/assets/5270903d-f9f7-4f1f-b59a-9f0229684e03" /> |

#### Test performed
- Test with trunk to verify the problem.
- Test with this PR to verify the fix.

#### Test environment
<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.9.0-dev
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.9.2
Log Directory Writable: ✔
WP Version: 6.8-RC2-60118
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.25.4
PHP Version: 7.4.33
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 8.5.0
OpenSSL/3.1.4

SUHOSIN Installed: –
MySQL Version: 8.0.33
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 23
page: 7
post: 2
product: 18
product_variation: 7
revision: 18
wp_font_face: 36
wp_font_family: 12
wp_global_styles: 2
wp_navigation: 1
wp_template: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (2) ###

Laravel DD for Wordpress: by Peter Hegman – 1.0.1
WooCommerce: by Automattic – 9.9.0-dev

### Inactive Plugins (2) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.7
Hello Dolly: by Matt Mullenweg – 1.7.2

### Must Use Plugins (1) ###

test.php: by  –

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: VND (₫)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 33 KB

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/ -  Contains the woocommerce/cart block
Checkout: #9 - /checkout/ -  Contains the woocommerce/checkout block
My account: #10 - /my-account/ -  Contains the [woocommerce_my_account] shortcode
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.6.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
Theme type: Classic theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store
add-to-cart-with-options-stepper-layout
disable-core-profiler-fallback

Disabled Features: product-data-views
experimental-blocks
coming-soon-newsletter-template
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
use-wp-horizon
blockified-add-to-cart

Daily Cron: ✔ Next scheduled: 2025-04-04 01:26:39 +00:00
Options: ✔
Notes: 64
Onboarding: completed

### Action Scheduler ###

Complete: 152
Oldest: 2025-03-06 01:28:36 +0000
Newest: 2025-04-03 09:26:37 +0000

Failed: 2
Oldest: 2025-03-06 01:28:36 +0000
Newest: 2025-03-18 00:28:50 +0000

Pending: 1
Oldest: 2025-04-04 09:24:06 +0000
Newest: 2025-04-04 09:24:06 +0000


### Status report information ###

Generated at: 2025-04-03 10:05:59 +00:00
```
</details>
